packages/utilities/src/warn.ts

```diff
@@ -0,0 +1,63 @@
+let _warningCallback = _warn;
+
+export interface IStringMap {
```

Comment:
Can we improve typing here? Something like this:

```TypeScript
type ISettingsMap<T> = {
    [P in keyof T]?: string;
};

function foo<T>(object: T, deprecations: ISettingsMap<T>) {
    for (let x in deprecations) {
        alert(deprecations[x]);
    }
}

////////

interface IX {
    foo: number;
    bar: string;
};

var x: IX = {
    foo: 42,
    bar: "test"
};

foo<IX>(x, {
    "foo": "abc"
});
```

(see http://www.typescriptlang.org/play/index.html#src=type%20ISettingsMap%3CT%3E%20%3D%20%7B%0D%0A%20%20%20%20%5BP%20in%20keyof%20T%5D%3F%3A%20string%3B%0D%0A%7D%3B%0D%0A%0D%0Afunction%20foo%3CT%3E(object%3A%20T%2C%20deprecations%3A%20ISettingsMap%3CT%3E)%20%7B%0D%0A%20%20%20%20for%20(let%20x%20in%20deprecations)%20%7B%0D%0A%20%20%20%20%20%20%20%20alert(deprecations%5Bx%5D)%3B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0A%2F%2F%2F%2F%2F%2F%2F%2F%0D%0A%0D%0Ainterface%20IX%20%7B%0D%0A%20%20%20%20foo%3A%20number%3B%0D%0A%20%20%20%20bar%3A%20string%3B%0D%0A%7D%3B%0D%0A%0D%0Avar%20x%3A%20IX%20%3D%20%7B%0D%0A%20%20%20%20foo%3A%2042%2C%0D%0A%20%20%20%20bar%3A%20%22test%22%0D%0A%7D%3B%0D%0A%0D%0Afoo%3CIX%3E(x%2C%20%7B%0D%0A%20%20%20%20%22foo%22%3A%20%22abc%22%0D%0A%7D)%3B)